provider/win_driver_installer_test.py

```diff
@@ -271,3 +276,259 @@ def iozone_test(test, params, vm, images):
             iozone.run(params['iozone_cmd_opitons'] % drive_letter)
         finally:
             iozone.clean()
+
+
+@error_context.context_aware
+def get_viofs_exe(test, params, session):
+    """
+    Get viofs.exe from virtio win iso,such as E:\viofs\2k19\amd64
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param session: the seesion of guest
+    """
+    error_context.context("Get virtiofs exe full path.", test.log.info)
+    media_type = params["virtio_win_media_type"]
+    try:
+        get_drive_letter = getattr(virtio_win, "drive_letter_%s" % media_type)
+        get_product_dirname = getattr(virtio_win,
+                                      "product_dirname_%s" % media_type)
+        get_arch_dirname = getattr(virtio_win, "arch_dirname_%s" % media_type)
+    except AttributeError:
+        test.error("Not supported virtio win media type '%s'", media_type)
+    viowin_ltr = get_drive_letter(session)
+    if not viowin_ltr:
+        test.error("Could not find virtio-win drive in guest")
+    guest_name = get_product_dirname(session)
+    if not guest_name:
+        test.error("Could not get product dirname of the vm")
+    guest_arch = get_arch_dirname(session)
+    if not guest_arch:
+        test.error("Could not get architecture dirname of the vm")
+
+    exe_middle_path = ("{name}\\{arch}" if media_type == "iso"
+                       else "{arch}\\{name}").format(name=guest_name,
+                                                     arch=guest_arch)
+    exe_file_name = "virtiofs.exe"
+    exe_find_cmd = 'dir /b /s %s\\%s | findstr "\\%s\\\\"'
+    exe_find_cmd %= (viowin_ltr, exe_file_name, exe_middle_path)
+    exe_path = session.cmd(exe_find_cmd).strip()
+    test.log.info("Found exe file '%s'", exe_path)
+    return exe_path
+
+
+def viofs_svc_create(test, params, session):
+    """
+    Only for windows guest, to create a virtiofs service
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param session: the seesion of guest
+    """
+    viofs_sc_create_cmd_default = 'sc create VirtioFsSvc binpath="%s" ' \
+                                  'start=auto  ' \
+                                  'depend="WinFsp.Launcher/VirtioFsDrv" ' \
+                                  'DisplayName="Virtio FS Service"'
+    viofs_sc_create_cmd = params.get("viofs_sc_create_cmd",
+                                     viofs_sc_create_cmd_default)
+    test.log.info("Register virtiofs service in Windows guest.")
```

Comment:
> According to the implementation of this function, I think it is creating the virtiofs service. So I am a bit of confuse about why it is registering.

"Start" is NOT good at here, since we have a related function.   
I prefer to use "create" at here if you think "register" is not a perfect option.

Reply:
> > According to the implementation of this function, I think it is creating the virtiofs service. So I am a bit of confuse about why it is registering.
> 
> "Start" is NOT good at here, since we have a related function. I prefer to use "create" at here if you think "register" is not a perfect option.

Yes, "create" is okay for me. I made a mistake with the typo, actually, I also want to "create" at that time :)

Reply:
I got you :)

> > > According to the implementation of this function, I think it is creating the virtiofs service. So I am a bit of confuse about why it is registering.
> > 
> > 
> > "Start" is NOT good at here, since we have a related function. I prefer to use "create" at here if you think "register" is not a perfect option.
> 
> Yes, "create" is okay for me. I made a mistake with the typo, actually, I also want to "create" at that time :)

I got you :)